### PR TITLE
Ees 2403

### PR DESCRIPTION
**Pull request by luke-h1**
This PR:
- Creates a custom locator so that test suites can use `testid:<test_id>` syntax instead of `//*[@data-testid="MyTestId"]`

### CODE REVIEW

File: tests/robot-tests/tests/admin/bau/prerelease.robot [hunk 1]
```diff
@@ -48,7 +48,7 @@ user chooses location, time period and filters
 
 user validates table rows
     user waits until results table appears     180
-    user waits until element contains  css:[data-testid="dataTableCaption"]
+    user waits until element contains  testid:datawTableCaption
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/prerelease.robot:51]: typo here: `datawTableCaption` -> `dataTableCaption`

File: tests/robot-tests/tests/admin/bau/create_data_block_with_chart.robot [hunk 2]
```diff
@@ -14,7 +14,6 @@ ${TOPIC_NAME}               %{TEST_TOPIC_NAME}
 ${PUBLICATION_NAME}         UI tests - create data block with chart %{RUN_IDENTIFIER}
 ${DATABLOCK_NAME}           UI test data block
 ${CONTENT_SECTION_NAME}     Test data block section
-
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin/bau/create_data_block_with_chart.robot:17]: Rogue line removal

File: tests/robot-tests/tests/admin_and_public/bau/prerelease_visibility.robot [hunk 3]
```diff
@@ -31,7 +31,7 @@ Go to 'Sign Off' page
     [Tags]  HappyPath
     user clicks link  Sign off
     user waits for page to finish loading
-    user waits until page contains testid  public-release-url
+    user waits until page contains  testid:public-release-url
```
Comment (markhiveit) [inline tests/robot-tests/tests/admin_and_public/bau/prerelease_visibility.robot:34]: this keyword should be `user waits until page contains element`

File: tests/robot-tests/tests/general_public/table_tool_exclusions_by_geographic_level.robot [hunk 4]
```diff
@@ -26,7 +26,6 @@ Validate "Exclusions by geographic level" subject details
     ${details}=  user gets details content element  More details  css:[data-testid="Radio item for Exclusions by geographic level"]
     user checks summary list contains  Geographic levels  Local Authority; National; Regional  ${details}
     user checks summary list contains  Time period  2006/07 to 2016/17  ${details}
-
```
Comment (markhiveit) [inline tests/robot-tests/tests/general_public/table_tool_exclusions_by_geographic_level.robot:29]: Rogue line removal

File: tests/robot-tests/tests/libs/common.robot [hunk 5]
```diff
@@ -23,7 +23,20 @@ do this on failure
     capture large screenshot and html
     set selenium timeout  3
 
+custom testid locator strategy
+  [Arguments]  ${browser}  ${test_id}   ${tag}  ${constraints}
+  Log to console  ${tag}
+  Log to console  ${constraints}
+  Log to console  ${test_id}
+  ${element}=   Execute Javascript  let xPathResult = document.evaluate('//*[@data-testid="${test_id}"]', document); if(xPathResult) return xPathResult.iterateNext(); return [];
+  user waits until element is visible  ${element}  30
```
Comment (markhiveit) [inline tests/robot-tests/tests/libs/common.robot:32]: This wait can be removed I think - I think the wait would need to be before the element is fetched with `execute javascript`. If we've retrieved the element already, there is no need to wait for it here, I believe

File: tests/robot-tests/tests/libs/common.robot [hunk 6]
```diff
@@ -23,7 +23,20 @@ do this on failure
     capture large screenshot and html
     set selenium timeout  3
 
+custom testid locator strategy
+  [Arguments]  ${browser}  ${test_id}   ${tag}  ${constraints}
+  Log to console  ${tag}
+  Log to console  ${constraints}
+  Log to console  ${test_id}
```
Comment (markhiveit) [inline tests/robot-tests/tests/libs/common.robot:30]: These log statements can be removed

**Review by markhiveit — CHANGES_REQUESTED** [review]
I think some of the errors I've found would fail on a test run - so I'd run the tests to ensure they're passing.

Also, it should now be possible to remove the `user clicks testid element` keyword, if it is no longer being used in test suites

File: tests/robot-tests/tests/libs/common.robot (hunk 6 above)
Comment (luke-h1) [reply tests/robot-tests/tests/libs/common.robot:30]: Cool, I left that in just in case this selector became flaky

**Review by markhiveit — CHANGES_REQUESTED** [review]
I think the only thing remaining is to remove `user clicks testid element` from common.robot

**Review by markhiveit — CHANGES_REQUESTED** [review]
I've also noticed that there are a few  `css:[data-testid="blah"]` that can be replaced to use `testid:`

**Comment by luke-h1** [comment]
Some tests seems to fail when using this new selector. For now I've left some instances of `css:[data-testid=""]` across the test suite.

**Review by markhiveit — APPROVED** [review]

**Pull request merged**